### PR TITLE
Check path to executable for `gensim.utils.check_output`. Fix #1485

### LINKED ISSUE #1485: proper check for filename before calling subprocess
Based on discussion [here](https://groups.google.com/forum/#!topic/gensim/nKVWoftW71Q), we should properly check for filename before `process calls` in `utils.check-output()`, and raise more intuitive exception and error message.

cc @jayantj

### PR DESCRIPTION
**Pull request by AverageS**
Add checks for kwargs, removed wrong Popen call.

### CODE REVIEW

File: gensim/test/test_utils.py [hunk 1]
```diff
@@ -83,6 +90,11 @@ def test_decode_entities(self):
         expected = u'It\x92s the Year of the Horse. YES VIN DIESEL \U0001f64c \U0001f4af'
         self.assertEquals(utils.decode_htmlentities(body), expected)
 
+    def test_check_output(self):
+        if os.name == 'posix':
```
Comment (menshikh-iv) [inline gensim/test/test_utils.py]: What's about windows here?

File: gensim/test/test_fasttext_wrapper.py [hunk 2]
```diff
@@ -30,8 +30,7 @@ def testfile():
 
 class TestFastText(unittest.TestCase):
     def setUp(self):
-        ft_home = os.environ.get('FT_HOME', None)
-        self.ft_path = os.path.join(ft_home, 'fasttext') if ft_home else None
+        self.ft_path = os.environ.get('FT_HOME', None)
```
Comment (menshikh-iv) [inline gensim/test/test_fasttext_wrapper.py]: unrelated change

File: gensim/utils.py [hunk 3]
```diff
@@ -1168,22 +1175,28 @@ def check_output(stdout=subprocess.PIPE, *popenargs, **kwargs):
...
-            error = subprocess.CalledProcessError(retcode, cmd)
-            error.output = output
-            raise error
-        return output
-    except KeyboardInterrupt:
-        process.terminate()
-        raise
+    logger.debug("COMMAND: %s %s", popenargs, kwargs)
+    command = kwargs.get('args')[0]
+    usr_bin_alias = '/usr/bin/' + command
```
Comment (menshikh-iv) [inline gensim/utils.py]: What's about windows case too?

File: gensim/utils.py [hunk 4]
```diff
@@ -1168,22 +1175,28 @@ def check_output(stdout=subprocess.PIPE, *popenargs, **kwargs):
...
+    if not os.path.exists(command) and os.path.exists(usr_bin_alias) and os.path.isfile(usr_bin_alias):
+        command = usr_bin_alias
+    if not os.path.exists(command):
```
Comment (menshikh-iv) [inline gensim/utils.py]: Need to add tests (system independent) for all raises case.

**Comment by menshikh-iv** [comment]
I close this PR because this looks abandoned (my bad).
@AverageS feel free to re-open if you want to continue.

File: gensim/test/test_fasttext_wrapper.py (hunk 2 above)
Comment (AverageS) [reply gensim/test/test_fasttext_wrapper.py]: small refactoring by the way. Looks a bit prettier.

File: gensim/test/test_utils.py (hunk 1 above)
Comment (AverageS) [reply gensim/test/test_utils.py]: It's about check_output

File: gensim/utils.py (hunk 3 above)
Comment (AverageS) [reply gensim/utils.py]: You've got a real problem here in 1173 row. If popenargs variable would not be empty then python will give an error about positional variable goes after named variable.

File: gensim/utils.py (hunk 4 above)
Comment (AverageS) [reply gensim/utils.py]: About the windows problem: as far as i remember, we ( me and @tmylk, Moscow, Summer 2017 you were there too :) ) decided that there are so many operating systems that it would not be possible to add functionality for testing all of them.

**Comment by AverageS** [comment]
Request to reopen and discuss

File: gensim/test/test_utils.py (hunk 1 above)
Comment (menshikh-iv) [reply gensim/test/test_utils.py]: `check_output` should works fine for windows too

File: gensim/utils.py (hunk 4 above)
Comment (menshikh-iv) [reply gensim/utils.py]: Times change :) Now we can test it with Appveyor (Windows). I see many users who use `gensim.models.wrappers` (i.e. use `check_output` implicitly) on windows, for this reason, I really worried about win-compatibility in this PR.

**Comment by menshikh-iv** [comment]
@AverageS also, please resolve merge-conflict & merge fresh `develop` to your branch.

**Comment by menshikh-iv** [comment]
Ping @AverageS, when do you plan to complete this PR?

**Comment by menshikh-iv** [comment]
This abandoned again, that's sad.

**Comment by DavidLeifer** [comment]
Just so I have it clear, you want a more descriptive error message to appear when the FastText wrapper is pointing at the FastText dir and not the executable, throwing this error: OSError: [Errno 13] Permission denied

But this doesn't work because it's not compatible with windows?

**Comment by menshikh-iv** [comment]
@DavidLeifer yes, but I know that some users use this feature on Windows too.

**Comment by DavidLeifer** [comment]
@menshikh-iv I tried to install make for Windows 10 from [http://gnuwin32.sourceforge.net/packages/make.htm] to build fastText so I can start testing. But I am having problems calling make, it says the command doesn't exist. Is there an easier way to do this?

**Comment by menshikh-iv** [comment]
@DavidLeifer as I remember, fasttext have unofficial read-to-use `.exe` binaries https://github.com/xiamx/fastText/releases, you can use binaries (instead of compiling it yourself).
